benchmarks/infra/stage-1/modules/gke-infra/cluster.tf

```diff
@@ -0,0 +1,249 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  _cluster_sa = (
+    local.cluster_create
+    ? module.cluster-service-account.0.email
+    : data.google_container_cluster.cluster.0.node_config.0.service_account
+  )
+  cluster_sa = (
+    local._cluster_sa == "default"
+    ? module.project.service_accounts.default.compute
+    : local._cluster_sa
+  )
+  cluster_sa_roles = [
+    "roles/artifactregistry.reader",
+    "roles/logging.logWriter",
+    "roles/monitoring.metricWriter",
+    "roles/monitoring.viewer",
+    "roles/stackdriver.resourceMetadata.writer"
+  ]
+  cluster_vpc = (
+    local.use_shared_vpc || !local.vpc_create
+    # cluster variable configures networking
+    ? {
+      network = try(
+        var.cluster_create.vpc.id, null
+      )
+      secondary_range_names = try(
+        var.cluster_create.vpc.secondary_range_names, null
+      )
+      subnet = try(
+        var.cluster_create.vpc.subnet_id, null
+      )
+    }
+    # VPC creation configures networking
+    : {
+      network               = module.vpc.0.id
+      secondary_range_names = { pods = "pods", services = "services" }
+      subnet                = values(module.vpc.0.subnet_ids)[0]
+    }
+  )
+}
+
+data "google_container_cluster" "cluster" {
+  count    = !local.cluster_create ? 1 : 0
+  project  = var.project_id
+  location = var.region
+  name     = var.cluster_name
+}
+
+module "cluster-service-account" {
+  source     = "git::https://github.com/GoogleCloudPlatform/cloud-foundation-fabric.git//modules/iam-service-account"
+  count      = local.cluster_create ? 1 : 0
+  project_id = module.project.project_id
+  name       = var.prefix
+}
+
+module "cluster-standard" {
+  source              = "git::https://github.com/GoogleCloudPlatform/cloud-foundation-fabric.git//modules/gke-cluster-standard"
```

Comment:
these are the same modules /infrastructure uses. let's not duplicate code. can this PR be updated to use /infrastructure?